### PR TITLE
kernel: add CONFIG_KSU_DEBUG

### PR DESCRIPTION
**Pull request by Ylarod**

### CODE REVIEW

**Comment by tiann** [comment]
why `shell` should be always allowed? App with wireless debug can acquire root with this feature

**Comment by Ylarod** [comment]
1. _Wireless Debug_ needs to be enabled by user manually, the users using KernelSU should know the risk of giving adb permission to apps.
2. Developers can have access to `su` without installing the manager by `adb shell`

**Comment by tiann** [comment]
> 1. _Wireless Debug_ needs to be enabled by user manually, the users using KernelSU should know the risk of giving adb permission to apps.
> 2. Developers can have access to `su` without installing the manager by `adb shell`

Users don't know giving adb permission would escape to root silently.

**Comment by Ylarod** [comment]
if KERNEL_SU_ALLOWLIST does not exist, allow `shell` by default

**Comment by tiann** [comment]
I think this should be a kernel CONFIG, if developer need this, he can open the config and build the kernel. normal users shouldn't have this feature.

**Comment by Ylarod** [comment]
maybe the `ksu_config` in #61 can be used in debug mode

File: kernel/allowlist.c [hunk 1]
```diff
@@ -181,7 +181,16 @@ void do_load_allow_list(struct work_struct *work)
 	fp = filp_open(KERNEL_SU_ALLOWLIST, O_RDONLY, 0);
 
 	if (IS_ERR(fp)) {
+#ifdef CONFIG_KSU_DEBUG
+		int errno = PTR_ERR(fp);
+		if (errno == -ENOENT) {
```
Comment (tiann) [inline kernel/allowlist.c:186]: Do we need to check errno here?

File: kernel/allowlist.c [hunk 2]
```diff
@@ -181,7 +181,16 @@ void do_load_allow_list(struct work_struct *work)
...
+			ksu_allow_uid(2000, true); // allow adb shell by default
+		}else{
```
Comment (tiann) [inline kernel/allowlist.c:188]: This code is not correctly formatted.

**Review by tiann — CHANGES_REQUESTED** [review]

File: kernel/allowlist.c (hunk 1 above)
Comment (Ylarod) [reply kernel/allowlist.c:186]: allow it only when KERNEL_SU_ALLOWLIST does not exist

Comment (tiann) [reply kernel/allowlist.c:186]: seems reasonable

**Comment by tiann** [comment]
Can we make the expected_size and expected_hash a kernel parameter in debug mode?

**Comment by Ylarod** [comment]
Not tested, let me try it first

**Comment by Ylarod** [comment]
```
rubens:/ # find /sys -name "ksu_*"
/sys/module/apk_sign/parameters/ksu_expected_hash
/sys/module/apk_sign/parameters/ksu_expected_size
```

**Review by tiann — APPROVED** [review]

**Comment by Ylarod** [comment]
```
rubens:/ # find /sys -name "ksu_*"
/sys/module/kernelsu/parameters/ksu_expected_hash
/sys/module/kernelsu/parameters/ksu_expected_size
```

**Comment by tiann** [comment]
looks great!

**Pull request merged**